src/lib.rs

```diff
@@ -41,26 +33,35 @@ impl Drop for BuildDir {
 }
 
 fn cp_r(in_dir: &path::Path, out_dir: &path::Path) {
-    let res = process::Command::new("cp")
-        .arg("-r")
-        .arg(in_dir)
-        .arg(out_dir)
-        .stdout(process::Stdio::inherit())
-        .stderr(process::Stdio::inherit())
-        .output()
-        .expect(&format!(
-            "Failed to cp -r {} {}",
-            in_dir.display(),
-            out_dir.display()
-        ));
-
-    assert!(
-        res.status.success(),
-        "Failed to cp -r {} {} with {:?}",
-        in_dir.display(),
-        out_dir.display(),
-        res
-    );
+    for entry in in_dir.read_dir().expect("read_dir call failed") {
+        let entry = entry.expect("Cannot access directory entry");
+        let entry_type = entry.file_type().expect("Cannot get directory entry file type");
+        let entry_path = entry.path();
+
+        let mut out_entry_path = out_dir.to_path_buf();
+        out_entry_path.push(entry.file_name());
+
+        if entry_type.is_dir() {
+            fs::create_dir_all(out_entry_path.clone()).expect("Cannot create directory");
+            cp_r(&entry_path, &out_entry_path);
+        } else {
+            // Check for potential conflict
+            if out_entry_path.exists() {
```

Comment:
Is this a case you encountered? It seems like it shouldn't happen, since the target is always uniquely named per build.

Reply:
(I'm always wary of file deletion code in anything automated)